.github/workflows/build.yml

```diff
@@ -1823,34 +221,11 @@ jobs:
           fetch-depth: 1
 
       - name: Gather information
-        shell: bash
-        run: |
-          # Gather values
-          SHORT_SHA=$(echo ${GITHUB_SHA} | cut -c1-8)
-          BUILD_ID='${{ github.job }}-${{ matrix.ci_lib_type }}'
-          CACHE_ID="${{ env.ccache_prefix }}-$BUILD_ID"
-          HASH_THIRDPARTY=$(cmake -DDIRECTORY_PATH="${{ env.ci_source_dir }}/Source/ThirdParty" -DHASH_FORMAT=short -P "${{ env.ci_source_dir }}/CMake/Modules/GetThirdPartyHash.cmake" 2>&1)
-          HASH_TOOLS=$(cmake -DDIRECTORY_PATH="${{ env.ci_source_dir }}/Source/Tools" -DHASH_FORMAT=short -P "${{ env.ci_source_dir }}/CMake/Modules/GetThirdPartyHash.cmake" 2>&1)
-          ARTIFACT_ID_SDK="rbfx-$BUILD_ID-$HASH_THIRDPARTY"
-          ARTIFACT_ID_TOOLS="rbfx-Linux-Tools-$HASH_TOOLS-$HASH_THIRDPARTY"
+        run: ${{ env.ci_build_script }} gather-info
 
-          # Display values
-          echo "SHORT_SHA=$SHORT_SHA"
-          echo "BUILD_ID=$BUILD_ID"
-          echo "CACHE_ID=$CACHE_ID"
-          echo "HASH_THIRDPARTY=$HASH_THIRDPARTY"
-          echo "HASH_TOOLS=$HASH_TOOLS"
-          echo "ARTIFACT_ID_SDK=$ARTIFACT_ID_SDK"
-          echo "ARTIFACT_ID_TOOLS=$ARTIFACT_ID_TOOLS"
-
-          # Save values to environment
-          echo "SHORT_SHA=$SHORT_SHA" >> $GITHUB_ENV
-          echo "BUILD_ID=$BUILD_ID" >> $GITHUB_ENV
-          echo "CACHE_ID=$CACHE_ID" >> $GITHUB_ENV
-          echo "HASH_THIRDPARTY=$HASH_THIRDPARTY" >> $GITHUB_ENV
-          echo "HASH_TOOLS=$HASH_TOOLS" >> $GITHUB_ENV
-          echo "ARTIFACT_ID_SDK=$ARTIFACT_ID_SDK" >> $GITHUB_ENV
-          echo "ARTIFACT_ID_TOOLS=$ARTIFACT_ID_TOOLS" >> $GITHUB_ENV
+      - name: Set up butler
+        if: ${{ (env.ci_execute_services == 'true') && (vars.DISABLE_ITCH != 'true') }}
```

Comment:
`env.ci_execute_services` is a bit too abstract. This is the flag to control whether we publish our shit, no? Let's call it this way then. `env.ci_publish` or whatever.

Reply:
It is for things like documentation and web hooks. Stuff specific to our repo that forks definitely dont want to run

Reply:
Yeah, but `ci_execute_services` is still too abstract. I can live with it, but I feel like it can be better.
"documentation" also counts as "publish". web hooks too, with some mental gymnastics.
we are publishing this build into the outer internet, one way or another.